pkg/security/security.go

```diff
@@ -351,12 +351,37 @@ type Caller struct {
 	Identities []string
 }
 
+type AuthContext struct {
+	// RequestContext is the context from request.
+	RequestContext context.Context
+	// Authenticators is the list of authenticators that were executed before in the auth chain.
+	Authenticators map[string]*Caller
+}
+
 type Authenticator interface {
-	Authenticate(ctx context.Context) (*Caller, error)
+	Authenticate(ctx AuthContext) (*Caller, error)
 	AuthenticatorType() string
 	AuthenticateRequest(req *http.Request) (*Caller, error)
 }
 
+func NewAuthContext(ctx context.Context) AuthContext {
+	return AuthContext{RequestContext: ctx, Authenticators: make(map[string]*Caller)}
+}
+
+func (ac *AuthContext) AddAuthenticator(authenticator string, caller *Caller) {
+	ac.Authenticators[authenticator] = caller
+}
+
+func (ac *AuthContext) Identities() []string {
```

Comment:
Not sure this is correct - for xfcc we need to only return the one from xfcc. 

Another option is to return separate identities:
- remote endpoint identity -  from xfcc, jwt or peer cert (in this order)
- connection peer identity - from socket cert or jwt or IP (if cidr range is used)
- user identity - for the case where both mtls and jwt are used, the mtls cert identifies the backend calling and jwt the actual user.